include/seqan3/alphabet/union_alphabet.hpp

```diff
@@ -129,13 +196,48 @@ struct union_alphabet
     //! the type of the alphabet when converted to char (e.g. via @link to_char @endlink)
     using char_type = typename first_alphabet_type::char_type;
     //! the type of the alphabet when represented as a number (e.g. via @link to_integral @endlink)
-    using integral_type = decltype(detail::at_least_type<value_size>());
+    using integral_type = detail::at_least_type_t<value_size>;
+    /*! the type used to assign a value from one of the base alphabets during
+     * copy construction or copy assignment. (i.e. via
+     * @link union_alphabet(const variant_type & alphabet) @endlink or
+     * @link operator= (const variant_type & alphabet) @endlink )
+     */
+    using variant_type = std::variant<first_alphabet_type, alphabet_types...>;
+
+    //! @name default constructors
+    //!@{
+    constexpr union_alphabet() = default;
+    constexpr union_alphabet(const union_alphabet &) = default;
+    constexpr union_alphabet(union_alphabet &&) = default;
+    constexpr union_alphabet(const integral_type & value) : value{value}{}
```

Comment:
How can I prevent that? Does a explicit constructor solve this? The integral_type can be `bool`, `uint8_t`, `uint16_t`, `uint32_t` or `uint64_t`.